File: Infrastructure/Services/ImageService.cs
using ApplicationCore.Entities;
using ApplicationCore.Identity;
using ApplicationCore.Interfaces;
using ApplicationCore.Interfaces.Services;
using Infrastructure.Data;
using Microsoft.Extensions.Configuration;
using System.Collections.Generic;

namespace Infrastructure.Services
{
    public abstract class ImageService<TImage, TEntity>
       : FileInfoService<TImage, TEntity>,
        IImageService<TImage, TEntity>
        where TImage : Image<TEntity>
        where TEntity : Entity
    {
        protected override IEnumerable<string> _allowedContentTypes { get; } = new List<string>() { @"image/jpeg", @"image/jpg", @"image/png" };

        public ImageService(
            StoreContext context,
            IIdentityService identityService,
            IScopedParameters scopedParameters,
            IAuthorizationParameters<TImage> authoriationParameters,
            IConfiguration configuration,
            IAppLogger<Service<TImage>> logger)
           : base(context, identityService, scopedParameters, authoriationParameters, configuration, logger)
        {
        }
    }
}

File: OctopusStore/ClientApp/src/app/item-image/image.service.ts
import { HttpClient } from '@angular/common/http';
import { FileInfoService } from './file-info.service';
import { Router } from '@angular/router';
import { Image } from './image';
import { Entity } from '../models/entity/entity';
import { IdentityService } from '../identity/identity.service';
import { MessageService } from '../message/message.service';

export abstract class ImageService<TEntity extends Entity> extends FileInfoService<Image<TEntity>> {

  constructor(
    protected http: HttpClient,
    protected router: Router,
    protected identityService: IdentityService,
    protected messageService: MessageService) {
    super(http, router, identityService, messageService);
    this.serviceName = 'Image service';
  }
}
